File: scratch/check_keys.js
const fs = require('fs');
const src = fs.readFileSync('Frontend/src/App.tsx', 'utf8');

const match = src.match(/const T: Record<Language, Record<string, string>> = (\{[\s\S]*?\n\});/);
const T = eval('(' + match[1] + ')');
const enKeys = new Set(Object.keys(T.en));

const accMatch = src.match(/const ACCOUNT_REVIEW_ORDER = (\[[\s\S]*?\]);/);
const kycMatch = src.match(/const KYC_REVIEW_ORDER = (\[[\s\S]*?\]);/);
const depMatch = src.match(/const DEPOSIT_REVIEW_ORDER = (\[[\s\S]*?\]);/);

const accOrders = eval(accMatch[1]);
const kycOrders = eval(kycMatch[1]);
const depOrders = eval(depMatch[1]);

console.log('Account orders missing from T:', accOrders.filter(k => !enKeys.has(k)));
console.log('KYC orders missing from T:', kycOrders.filter(k => !enKeys.has(k)));
console.log('Deposit orders missing from T:', depOrders.filter(k => !enKeys.has(k)));

const qKeys = Array.from(src.matchAll(/key:\s*"([^"]+)"/g)).map(m => m[1]);
const uniqueQKeys = [...new Set(qKeys)];
console.log('Question keys missing from T:', uniqueQKeys.filter(k => !enKeys.has(k)));

File: scratch/check_keys.py
import re

with open('Frontend/src/App.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

en_match = re.search(r'\ben:\s*\{([\s\S]*?)\n  \},', content)
en_keys = set(re.findall(r'([a-zA-Z0-9_]+)\s*:', en_match.group(1)))

q_keys = set(re.findall(r'key:\s*"([a-zA-Z0-9_]+)"', content))

missing = q_keys - en_keys
print('Question keys missing from T:', sorted(list(missing)))
